Add `:ref:` targets to `recwarn.rst`.

lwm opened the pull request:
Closes https://github.com/pytest-dev/pytest/issues/297.

[Greetings from Zagreb!](https://www.meetup.com/Python-Hrvatska/events/235507740/)

lwm commented:
CI failure seems unrelated ([see report](https://ci.appveyor.com/project/pytestbot/pytest/build/1.0.2073/job/wd80428foc222mhe#L81), given that https://github.com/pytest-dev/pytest/pull/2088 and https://github.com/pytest-dev/pytest/pull/2085 also had similar failures.

nicoddemus commented:
> CI failure seems unrelated (see report), given that #2088 and #2085 also had similar failures.

Agree, I will take a look.

nicoddemus commented:
Failure related to: pypa/setuptools#861

nicoddemus commented:
> Greetings from Zagreb!

Greetings! No wonder there's been so much activity today! 😆 

We really appreciate the efforts!

nicoddemus commented on CHANGELOG.rst:
```diff
@@ -1,6 +1,9 @@
 3.0.5.dev0
 ==========
 
+* Provide `:ref:` targets for `recwarn.rst` so we can use intersphinx referencing.
```
You should use double ` for mono space text in RST:

```
Provide ``:ref:`` targets for ``recwarn.rst`` so we can use intersphinx referencing.
```

lwm commented:
> >  Greetings from Zagreb!
>
> Greetings! No wonder there's been so much activity today! :laughing:
> We really appreciate the efforts!

Happy to help :beers: 

Fixed up `CHANGELOG.rst` based on your comment and rebased on master.

Thanks for clearing up issue with https://github.com/pytest-dev/pytest/pull/2092.

nicoddemus commented:
Nice! As soon as CI passes I will merge this!
